### DISCUSSION TITLE
Is it possible to use NextJS (Server Side) with N-API and node-addons (C++ code)?

### DISCUSSION THREAD

**Discussion by shreks7** [vercel/next.js#17807] [Help]
Hi,

I am trying to use C++ library with NextJS. Is it possible to use NAPI [Node-Addon](https://medium.com/jspoint/a-simple-guide-to-load-c-c-code-into-node-js-javascript-applications-3fcccf54fd32) during server-side rendering? 

Any pointers will be helpful! Thank you.

**Comment by sdasun** [comment]
@shreks7  you may already solved this. I'm writing this to help if anyone else having the same issue. 
short answer use `__non_webpack_require__` instead `require`. 

The issue is with the WebPack not with Next.js. In theory, if you add the module to the web pack externals (in the next.config.js file) this one should works. Yet, it didn't work for me.

**Comment by chanoo** [comment]
` yarn add -D copy-webpack-plugin`
```
const CopyPlugin = require('copy-webpack-plugin');
...
  webpack: (config) => {
    config.plugins.push(
      new CopyPlugin({
        patterns: [
          {
            from: './addon/helloworld/build/Release/hello.node',
            to: './addon/dist/hello.node',
          },
        ],
      })
    );
    return config;
  },
...
```



`yarn add -D @types/webpack-env                            `                

```
  const node = __non_webpack_require__('../../addon/dist/hello.node');
  console.log(node.hello()); // 'world'
```

**Reply by FangliangBai** [reply]
This solution works for me. Thank you!

**Reply by arichyx** [reply]
In this way typings of `hello.node` are lost. Any way to solve it?
